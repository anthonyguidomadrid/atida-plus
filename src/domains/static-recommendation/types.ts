import { SessionChannelType } from '~domains/basket/types'
import { Product } from '~domains/product'

export type StaticRecommendationDataState = {
  isLoading: boolean
  wasSuccess: boolean
  wasError: boolean
  error?: string
}

export type FetchStaticRecommendation = {
  key: string
  products: Partial<Product>[]
}

export type FetchStaticRecommendationTriggerPayload = {
  list: FetchStaticRecommendation[]
  sessionChannel?: SessionChannelType
}

export type FetchStaticRecommendationDataState = StaticRecommendationDataState & {
  details?: FetchStaticRecommendation[]
}
