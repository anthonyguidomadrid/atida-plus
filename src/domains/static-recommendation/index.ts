export * from './sagas'
export * from './slices'
export * from './selectors'
