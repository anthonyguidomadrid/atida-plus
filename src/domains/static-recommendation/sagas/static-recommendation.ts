import type { PayloadAction } from '@reduxjs/toolkit'
import type { SagaIterator } from 'redux-saga'
import { all, call, getContext, put, takeEvery } from 'redux-saga/effects'
import { logger } from '~helpers'
import {
  fetchStaticRecommendationTrigger,
  fetchStaticRecommendationRequest,
  fetchStaticRecommendationSuccess,
  fetchStaticRecommendationFailure,
  fetchStaticRecommendationFulfill
} from '../slices'

import { getErrorMessage } from '~helpers/error'
import {
  FetchStaticRecommendation,
  FetchStaticRecommendationTriggerPayload
} from '../types'

export const dispatchStaticRecommendationFetchRequest = async (
  data: FetchStaticRecommendationTriggerPayload,
  locale: string
): Promise<any> => {
  const {
    fetchStaticRecommendation
    // eslint-disable-next-line @typescript-eslint/no-var-requires
  } = require('../services/static-recommendation')
  return fetchStaticRecommendation(locale, data)
}

function* doFetchStaticRecommendationSaga({
  payload
}: PayloadAction<FetchStaticRecommendationTriggerPayload>) {
  try {
    yield put(fetchStaticRecommendationRequest())
    const locale: string = yield getContext('locale')
    const apiResponse: FetchStaticRecommendation[] = yield call(
      dispatchStaticRecommendationFetchRequest,
      payload,
      locale
    )

    yield put(fetchStaticRecommendationSuccess(apiResponse))
  } catch (error) {
    const errorMessages = getErrorMessage(error)
    logger.error(errorMessages)
    if (Array.isArray(errorMessages)) {
      yield put(
        fetchStaticRecommendationFailure({
          message: errorMessages[0]
        })
      )
    } else {
      yield put(
        fetchStaticRecommendationFailure({
          message: errorMessages
        })
      )
    }
  } finally {
    yield put(fetchStaticRecommendationFulfill())
  }
}

export function* fetchStaticRecommendationSaga(): SagaIterator {
  yield all([
    takeEvery(fetchStaticRecommendationTrigger, doFetchStaticRecommendationSaga)
  ])
}
