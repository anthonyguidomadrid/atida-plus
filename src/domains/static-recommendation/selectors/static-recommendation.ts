import { createSelector } from '@reduxjs/toolkit'

import { RootState } from '~domains/redux'

const selectStaticRecommendation = (state: RootState, props: { id?: string }) =>
  state?.server?.staticRecommendation['static-recommendation']?.details?.find(
    item => item.key === props?.id
  )

export const selectStaticRecommendationProducts = createSelector(
  selectStaticRecommendation,
  products => products?.products
)
