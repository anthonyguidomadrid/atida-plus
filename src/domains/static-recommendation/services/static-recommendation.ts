import {
  ChannelPrice,
  ElasticSearchProduct,
  normalizeElasticSearchMinimalProduct,
  Product
} from '~domains/product'
import {
  createElasticSearchClient,
  getElasticSearchProductIndex
} from '~domains/elasticsearch/helpers/client'
import {
  FetchStaticRecommendation,
  FetchStaticRecommendationTriggerPayload
} from '../types'
import { FeatureFlag } from '~config/constants/feature-flags'
import { loadFeatureFlag } from '~helpers/server-only/featureFlagClient'

if (typeof window !== 'undefined')
  throw new Error('This file should not be called client-side')

export const fetchStaticRecommendation = async (
  locale: string,
  data: FetchStaticRecommendationTriggerPayload
): Promise<FetchStaticRecommendation[]> => {
  let skus: Partial<Product>[] = []

  data?.list?.map(value => {
    skus = [...skus, ...value.products]
  })

  skus = skus.filter((value, index, self) => self.indexOf(value) === index)

  try {
    // TODO: Remove when the price channels on products are ready to be used
    const mockedProductPriceChannels = (await loadFeatureFlag(
      locale,
      FeatureFlag.MUTLICHANNEL_PRICES_MOCKED_PRODUCT_PRICE_CHANNEL
    )) as { channel_prices: ChannelPrice[] }
    const elasticSearchClient = createElasticSearchClient()
    const elasticSearchResponse = await elasticSearchClient.mget<{
      docs: ElasticSearchProduct[]
    }>({
      index: getElasticSearchProductIndex(locale),
      body: {
        ids: skus
      }
    })

    const items = elasticSearchResponse.body.docs.filter(doc => doc.found)

    const elasticData = data?.list?.reduce<FetchStaticRecommendation[]>(
      (previousValue, currentValue) => {
        const { products } = currentValue
        const filteredItems = items.filter(item =>
          (products as string[]).includes(item._id)
        )
        return [
          ...previousValue,
          {
            ...currentValue,
            products: filteredItems.map(item =>
              normalizeElasticSearchMinimalProduct(
                locale,
                item,
                data?.sessionChannel,
                mockedProductPriceChannels?.channel_prices
              )
            )
          }
        ]
      },
      []
    )

    return elasticData
  } catch (error) {
    return data.list
  }
}
