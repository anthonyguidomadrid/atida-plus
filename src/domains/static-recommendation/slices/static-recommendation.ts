import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import { clearStateErrors, hydrate } from '~domains/redux/actions'
import { hasOwnProperty } from '~helpers'
import {
  FetchStaticRecommendation,
  FetchStaticRecommendationDataState,
  FetchStaticRecommendationTriggerPayload
} from '../types'

const initialState: FetchStaticRecommendationDataState = {
  isLoading: false,
  wasSuccess: false,
  wasError: false
}

export const fetchStaticRecommendationSlice = createSlice({
  name: 'Static Recommendation',
  initialState,
  reducers: {
    trigger(
      _state,
      _action: PayloadAction<FetchStaticRecommendationTriggerPayload>
    ) {
      // no-op, used for triggering saga
    },
    request(state) {
      state.isLoading = true
      state.wasSuccess = false
      state.wasError = false
      delete state.error
    },
    success(state, action: PayloadAction<FetchStaticRecommendation[]>) {
      state.details = action.payload
      state.wasSuccess = true
    },
    failure(state, action: PayloadAction<{ message: string }>) {
      const { message } = action.payload
      state.wasError = true
      state.error = message
    },
    fulfill(state) {
      state.isLoading = false
    }
  },

  extraReducers: builder => {
    builder.addCase(clearStateErrors, (state, _action) => {
      state.wasError = false
      delete state.error
    })
    builder.addCase(hydrate, (_state, action) => {
      if (
        typeof action.payload === 'object' &&
        hasOwnProperty(action.payload, 'staticRecommendation') &&
        typeof action.payload.staticRecommendation === 'object' &&
        hasOwnProperty(
          action.payload.staticRecommendation,
          'static-recommendation'
        )
      ) {
        return action.payload.staticRecommendation[
          'static-recommendation'
        ] as FetchStaticRecommendationDataState
      }
    })
  }
})
