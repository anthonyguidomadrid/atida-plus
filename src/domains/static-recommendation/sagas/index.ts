import type { SagaIterator } from 'redux-saga'
import { all, fork } from 'redux-saga/effects'

import { fetchStaticRecommendationSaga } from './static-recommendation'

export function* staticRecommendationSaga(): SagaIterator {
  yield all([fetchStaticRecommendationSaga].map(fork))
}
