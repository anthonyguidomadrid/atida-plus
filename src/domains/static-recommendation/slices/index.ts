import { combineReducers } from '@reduxjs/toolkit'
import { fetchStaticRecommendationSlice } from '../slices/static-recommendation'

export const recommendationsReducer = combineReducers({
  'static-recommendation': fetchStaticRecommendationSlice.reducer
})

export const {
  trigger: fetchStaticRecommendationTrigger,
  request: fetchStaticRecommendationRequest,
  success: fetchStaticRecommendationSuccess,
  failure: fetchStaticRecommendationFailure,
  fulfill: fetchStaticRecommendationFulfill
} = fetchStaticRecommendationSlice.actions
