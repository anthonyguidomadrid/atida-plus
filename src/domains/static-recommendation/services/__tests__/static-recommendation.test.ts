/**
 * @jest-environment node
 */
// @ts-ignore
import { mget } from '@elastic/elasticsearch'
import { Product } from '~domains/product/types'
import {
  staticRecommendationPayload,
  staticRecommendationNormalizedDataResponse
} from '../../__mocks__/staticRecommendation'
import { fetchStaticRecommendation } from '../static-recommendation'

describe(fetchStaticRecommendation, () => {
  beforeEach(() => {
    ;(mget as jest.Mock).mockResolvedValue(staticRecommendationPayload)
  })

  it('passes the correct parameters to elasticsearch request', async () => {
    await fetchStaticRecommendation('en-gb', {
      list: [
        {
          key: staticRecommendationPayload.list[0].key,
          products: staticRecommendationPayload.list[0].products.map(
            product => product
          ) as Partial<Product>[]
        }
      ],
      sessionChannel: undefined
    })

    expect(mget).toHaveBeenCalledWith({
      index: 'index-en-gb',
      body: {
        ids: staticRecommendationPayload.list[0].products
      }
    })
  })

  it('returns the normalized product response - without session channel', async () => {
    const response = await fetchStaticRecommendation('en-gb', {
      list: [
        {
          key: staticRecommendationNormalizedDataResponse[0].key,
          products: staticRecommendationNormalizedDataResponse[0].products.map(
            product => product
          ) as Partial<Product>[]
        }
      ],
      sessionChannel: undefined
    })

    expect(response).toMatchSnapshot()
  })
  it('returns the normalized product response - with session channel', async () => {
    const response = await fetchStaticRecommendation('en-gb', {
      list: [
        {
          key: staticRecommendationNormalizedDataResponse[0].key,
          products: staticRecommendationNormalizedDataResponse[0].products.map(
            product => product
          ) as Partial<Product>[]
        }
      ],
      sessionChannel: { channel: 'amazon', sku: '100000001100000002100000003' }
    })

    expect(response).toMatchSnapshot()
  })
})
