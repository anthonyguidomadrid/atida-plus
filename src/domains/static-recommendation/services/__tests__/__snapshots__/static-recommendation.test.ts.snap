// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchStaticRecommendation returns the normalized product response - with session channel 1`] = `
Array [
  Object {
    "key": "100000001100000002100000003",
    "products": Array [
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000001",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000002",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000003",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
    ],
  },
]
`;

exports[`fetchStaticRecommendation returns the normalized product response - without session channel 1`] = `
Array [
  Object {
    "key": "100000001100000002100000003",
    "products": Array [
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000001",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000002",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
      Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
        },
        "availability": "AVAILABLE",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "labels": Array [
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "name": "Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas",
        "price": Object {
          "currency": "EUR",
          "value": 8888,
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "sku": "100000003",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas",
      },
    ],
  },
]
`;
